float64 x
float64 y
float64 d
float64 theta
float64 radius

